### ISSUE TITLE
Feature Request: Self-destructing messages

### ISSUE THREAD

**Issue by semekh** [RocketChat/feature-requests#131]
**Is your feature request related to a problem? Please describe.**
We use RocketChat as our internal organization communication platform. Every now and then, we need to exchange sensitive messages (e.g. passwords, personal information, etc.) right away, but we'd like to have those messages disappear automatically after they've served their purpose.

**Describe the solution you'd like**
It'd be great to be able to set a self-destruction timer. That way, in such cases, we'll mark the message as self-destructing, and just forget about it.

**Describe alternatives you've considered**
* Sending such messagses through other communication channels that have this features (e.g. Telegram, GMail) with the downside that one needs to have the other person's contact info on the alternative communication channel.
* Sending the message and remembering to delete them afterwards. Prone to human error.

**Comment by dantosXD** [comment]
Second

**Comment by bug** [comment]
+1

**Comment by oskarissimus** [comment]
+1 i can try to implement this if someone would guide me

**Comment by qx-775** [comment]
I think this adds **a lot** of value for companies to use in the free plan and this is definitely a reason for companies to switch from other communication services if this was securely available. 

### The problem
In organisations I often see too often passwords and personal data exchanged using 3rd party tools to temporary generate a link with some data or they even use MS Teams to communicate. But can we trust those? Are they open source? Most often not. I believe personal data and passwords should _never_ leave the company network. 

### How Rocket.Chat could generate business value and increase privacy of companies
With a self hosted rocket.chat instance where people can just open a chat to eachother and set a self destructing message that can be set with a customizable timer of 30s 1m 1h 24h or custom (like [Signal](signal.org) does)  that will self destruct after reading will be very handy.

### Help needed
necrobumping @oskarissimus question if he is still up for it, can someone guide him?

**Comment by oskarissimus** [comment]
yeah, im still up for it! And I still don't know where to start...

**Comment by atmovantage** [comment]
Want to keep this idea on the radar because I think it's becoming a popular feature on other platforms like Telegram and Gab. Would be a good addition for RC
